pkg/model/awsmodel/spotinst.go

```diff
@@ -341,30 +341,31 @@ func (b *SpotInstanceGroupModelBuilder) buildOcean(c *fi.ModelBuilderContext, ig
 		ocean.UseAsTemplateOnly = fi.Bool(true)
 	}
 
-	ig := igs[0].DeepCopy()
-	if len(igs) > 1 {
-		for _, g := range igs {
-			for k, v := range g.ObjectMeta.Labels {
-				if k == SpotInstanceGroupLabelOceanDefaultLaunchSpec {
-					defaultLaunchSpec, err := parseBool(v)
-					if err != nil {
-						continue
-					}
-					if fi.BoolValue(defaultLaunchSpec) {
-						if ig != nil {
-							return fmt.Errorf("unable to detect default launch spec: "+
-								"multiple instance groups labeled with `%s: \"true\"`",
-								SpotInstanceGroupLabelOceanDefaultLaunchSpec)
-						}
-						ig = g.DeepCopy()
-						break
+	var ig *kops.InstanceGroup
+	for _, g := range igs {
+		for k, v := range g.ObjectMeta.Labels {
+			if k == SpotInstanceGroupLabelOceanDefaultLaunchSpec {
+				defaultLaunchSpec, err := parseBool(v)
+				if err != nil {
+					continue
+				}
+				if fi.BoolValue(defaultLaunchSpec) {
+					if ig != nil {
+						return fmt.Errorf("unable to detect default launch spec: "+
+							"multiple instance groups labeled with `%s: \"true\"`",
+							SpotInstanceGroupLabelOceanDefaultLaunchSpec)
 					}
+					ig = g.DeepCopy()
+					break
 				}
 			}
 		}
-
-		klog.V(4).Infof("Detected default launch spec: %q", b.AutoscalingGroupName(ig))
 	}
+	if ig == nil {
+		ig = igs[0].DeepCopy()
```

Comment:
I would expect to return or error in this case,
```suggestion
		return nil
```

Reply:
I ok with that, but the current and pass behaviour was if you not set the label it will use the first instance group for the default launch spec

Reply:
Im ok with that, but the current and pass behaviour was if you not set the label it will use the first instance group for the default launch spec

Reply:
I looked again through the code and seems fine like this.